filesystem/iso9660.ksy

```diff
@@ -7,269 +7,959 @@ meta:
+            doc-ref: ecma-119 8.5.14
+            type: text128
+          - id: data_preparer_identifier
+            doc-ref: ecma-119 8.5.15
+            type: text128
+          - id: application_identifier
+            doc-ref: ecma-119 8.5.16
+            type: text128
+          - id: copyright_file_identifier
+            doc-ref: ecma-119 8.5.17
+            type: text37
+          - id: abstract_file_identifier
+            doc-ref: ecma-119 8.5.18
+            type: text37
+          - id: bibliographic_file_identifier
+            doc-ref: ecma-119 8.5.19
+            type: text37
+          - id: volume_creation_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: volume_modification_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: volume_expiration_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: volume_effective_date_and_time
+            doc-ref: ecma-119 8.5
+            type: datetime_long
+          - id: file_structure_version
+            doc-ref: ecma-119 8.5
+            type: s1
+        instances:
+          path_table:
+            io: _root._io
+            pos: _root.sector_size * loc_l_path_table
+            size: path_table_size.le
+            type: path_table_records
+      path_table_records:
+        doc-ref: ecma-119 9.4
+        seq:
+          - id: path_table_record
+            type: path_table_record
+            repeat: eos
+        types:
+          path_table_record:
+            doc-ref: ecma-119 9.4
+            seq:
+              - id: len_dir_name
+                doc-ref: ecma-119 9.4.1
+                type: u1
+              - id: len_ext_rec
+                doc-ref: ecma-119 9.4.2
+                type: u1
+              - id: loc_ext
+                doc-ref: ecma-119 9.4.3
+                type: u4le
+              - id: parent_dir_num
+                doc-ref: ecma-119 9.4.4
+                type: u2le
+              - id: dir_name
+                doc-ref: ecma-119 9.4.5
+                type: str
+                size: len_dir_name
+              - id: padding_field
+                doc-ref: ecma-119 9.4.6
+                doc: |
+                  Padding field is added when len_dir_name contains an odd number
+                contents: [ 0x0 ]
+                if: len_dir_name % 2 == 1
+            instances:
+              directory_records:
+                io: _root._io
+                pos: _root.sector_size * loc_ext
+                size: _root.sector_size
+                type: directory_record
+      directory_records:
+        doc: |
+          First item "." it points to it self
+          Second item ".." it points to the parent, or also to self if it the root
+        seq:
+          - id: directory_record
+            type: directory_record
+            repeat: until
+            repeat-until: _.len_dr == 0
```

Comment:
This is actually not correct, as there can be additional padding bytes. Section 6.8.1.1 of the ISO9660 spec says:

```
Each Directory  Record  shall  end  in  the  Logical  Sector  in  which  it  begins.  Unused  byte  positions  after  the  last Directory Record in a Logical Sector shall be set to (00).
```

So this means that you also need to look at the remaining bytes in the logical sector as well as the total length of the directory records.